lib/board.rb

```diff
@@ -0,0 +1,82 @@
+require('pry')
+
+class Board
+  attr_accessor :cells
+
+  WIN_COMBINATIONS = [
+    [0,1,2],
+    [3,4,5],
+    [6,7,8],
+    [0,3,6],
+    [1,4,7],
+    [2,5,8],
+    [0,4,8],
+    [2,4,6]
+  ]
+
+  def initialize
+    reset!
+  end
+
+  def won?
+    WIN_COMBINATIONS.each do |win|
+      if win.all? { |ind| cells[ind] =="X" } || win.all? { |ind|  cells[ind] =="O" }
+        return win
+      end
+    end
+    false
+  end
+
+  def full?
+    !self.cells.any?{|i| i==" "}
+  end
+
+  def draw?
+    !won? && full?
+  end
+
+  def winner
+    won? ? self.cells[winning_combo[0]] : nil
+  end
+
+  # Define display_board that accepts a board and prints
+  # out the current state.
+  def display
+    puts " #{self.cells[0]} | #{self.cells[1]} | #{self.cells[2]} "
+    puts "-----------"
+    puts " #{self.cells[3]} | #{self.cells[4]} | #{self.cells[5]} "
+    puts "-----------"
+    puts " #{self.cells[6]} | #{self.cells[7]} | #{self.cells[8]} "
+  end
+
+  def position(index)
+    self.cells[index.to_i-1]
+  end
+
+  def position_on_board?(index)
+    index.to_i.between?(1,9)
+  end
+
+  def turn_count
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```